pkg/scheduler/plugins/loadaware/load_aware.go

```diff
@@ -0,0 +1,304 @@
+			time.Since(nodeMetric.Status.UpdateTime.Time) >= time.Duration(nodeMetricExpirationSeconds)*time.Second
+}
+
+func (p *Plugin) estimatedAssignedPodUsage(nodeName string, nodeMetric *slov1alpha1.NodeMetric) map[corev1.ResourceName]int64 {
+	estimatedUsed := make(map[corev1.ResourceName]int64)
+	nodeMetricReportInterval := getNodeMetricReportInterval(nodeMetric)
+	p.podAssignCache.lock.RLock()
+	defer p.podAssignCache.lock.RUnlock()
+	for _, assignInfo := range p.podAssignCache.podInfoItems[nodeName] {
+		if assignInfo.timestamp.After(nodeMetric.Status.UpdateTime.Time) ||
+			assignInfo.timestamp.Before(nodeMetric.Status.UpdateTime.Time) &&
+				nodeMetric.Status.UpdateTime.Sub(assignInfo.timestamp) < nodeMetricReportInterval {
+			estimated := estimatedPodUsed(assignInfo.pod, p.args.ResourceWeights, p.args.EstimatedScalingFactors)
+			for resourceName, value := range estimated {
+				estimatedUsed[resourceName] += value
+			}
+		}
+	}
+	return estimatedUsed
+}
+
+func getNodeMetricReportInterval(nodeMetric *slov1alpha1.NodeMetric) time.Duration {
+	if nodeMetric.Spec.CollectPolicy == nil || nodeMetric.Spec.CollectPolicy.ReportIntervalSeconds == nil {
+		return DefaultNodeMetricReportInterval
+	}
+	return time.Duration(*nodeMetric.Spec.CollectPolicy.ReportIntervalSeconds) * time.Second
+}
+
+func estimatedPodUsed(pod *corev1.Pod, resourceWeights map[corev1.ResourceName]int64, scalingFactors map[corev1.ResourceName]int64) map[corev1.ResourceName]int64 {
+	requests, limits := resourceapi.PodRequestsAndLimits(pod)
+	estimatedUsed := make(map[corev1.ResourceName]int64)
+	priorityClass := extension.GetPriorityClass(pod)
+	for resourceName := range resourceWeights {
+		realResourceName := extension.TranslateResourceNameByPriorityClass(priorityClass, resourceName)
+		estimatedUsed[resourceName] = estimatedUsedByResource(requests, limits, realResourceName, scalingFactors[resourceName])
+	}
+	return estimatedUsed
+}
+
+func estimatedUsedByResource(requests, limits corev1.ResourceList, resourceName corev1.ResourceName, scalingFactor int64) int64 {
+	limitQuantity := limits[resourceName]
+	requestQuantity := requests[resourceName]
+	var quantity resource.Quantity
+	if limitQuantity.Cmp(requestQuantity) > 0 {
+		scalingFactor = 100
+		quantity = limitQuantity
+	} else {
+		quantity = requestQuantity
+	}
+
+	if quantity.IsZero() {
+		switch resourceName {
+		case corev1.ResourceCPU, extension.BatchCPU:
+			return DefaultMilliCPURequest
+		case corev1.ResourceMemory, extension.BatchMemory:
+			return DefaultMemoryRequest
+		}
+		return 0
+	}
+
+	var estimatedUsed int64
+	switch resourceName {
+	case corev1.ResourceCPU:
+		estimatedUsed = int64(math.Round(float64(quantity.MilliValue()) * float64(scalingFactor) / 100))
+		if estimatedUsed > limitQuantity.MilliValue() {
+			estimatedUsed = limitQuantity.MilliValue()
+		}
+	default:
+		estimatedUsed = int64(math.Round(float64(quantity.Value()) * float64(scalingFactor) / 100))
```

Comment:
How about keeping the type `resource.Quantity` until the final scoring to improve precisions?